### ISSUE TITLE
Enhancement: [consistent-type-imports] prevent type-only imports from using inline type specifiers

### ISSUE THREAD

**Issue by latin-1** [typescript-eslint/typescript-eslint#6338]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-imports

### Description

With `"importsNotUsedAsValues": "error"` in `tsconfig.json`, those two lines behave differently.

```ts
import type { T } from "./module"; // ok
import { type T } from "./module"; // error ts(1371)
```

While using the rule `consistent-type-imports` with option `"fixStyle": "inline-type-imports"`, this plugin will yield incorrect ts code.

```ts
// before
import { T } from "./module";

// after
import { type T } from "./module"; // error ts(1371)
```

We may need a new option or a new rule to handle the type-only imports properly.

```ts
import { type T }         -> import type { T }
import { type A, type B } -> import type { A, B }
import { A, type B }      -> <as-is>
import { A, B }           -> <as-is>
```

### Fail

```typescript
import { type T }
import { type A, type B }
```

### Pass

```typescript
import type { T }
import type { A, B }
import { A, type B }
import { A, B }
```

### Additional Info

https://github.com/microsoft/TypeScript/issues/47118

**Comment by bradzacher** [comment]
Have you tried using the `separate-type-imports` fix style? In that mode it should fix to top-level type imports, not inline type imports.

Failing that the rule you found implemented in https://github.com/import-js/eslint-plugin-import/issues/2469 would also do the job here to enforce top level imports.

There's also the new feature specced in https://github.com/import-js/eslint-plugin-import/issues/2470 which could help.

There's a lot of different things that you can use to get the style you want.
Changing or adding a new fixer isn't going to happen IMO because we already have two fixers that both do exactly what most people want (either inline or top level).

**Comment by latin-1** [comment]
#### https://github.com/import-js/eslint-plugin-import/issues/2469

This doesn't help. What I need is:
```ts
import type { T } from "./mod"; // type-only
import { type A, B } from "./mod"; // mixed
import { A, B } from "./mod"; // non-type
```
'consistent-type-specifier-style' forces all imports to use inline or non-inline type specifiers, which is not suitable for me.

#### https://github.com/import-js/eslint-plugin-import/issues/2470

Although it may solve my problem, I do not believe it is the best solution.
```ts
import { T, V } from "./mod";
// ↓ code fix by @typescript-eslint/consistent-type-imports
import type { T } from "./mod";
import { V } from "./mod";
// ↓ code fix by import/no-duplicates
import { type T, V } from "./mod";
```

**Comment by latin-1** [comment]
I'd prefer to use inline type specifiers. But it's not possible to use it for type-only imports due to TypeScript's design. So I'm looking for something to mix them up.

**Comment by latin-1** [comment]
>There's a lot of different things that you can use to get the style you want.

This is not something related to **style**. They have different **semantics** and they are **not interchangeable**.

```ts
import type { T } from "./mod";
```
compiles to
```ts
// empty
```

but
```ts
import { type T } from "./mod";
```
complies to
```ts
import {} from "./mod";
```

**Comment by latin-1** [comment]
This is the expected behavior of tsc. They **do** have different behavior.

https://github.com/microsoft/TypeScript/issues/47118#issuecomment-993046561
https://github.com/microsoft/TypeScript/issues/47118#issuecomment-993047937

**Comment by latin-1** [comment]
In fact, `import { type T }` **make no sense** if `importsNotUsedAsValues` is set to `"error"` and should be avoided in the code fix.

**Comment by latin-1** [comment]
@bradzacher What are your thoughts? Should we check if `importsNotUsedAsValues` is set to `"error"` in the code fix or provide another option? Or do you believe they should be included in eslint-plugin-import?

**Comment by bradzacher** [comment]
Using `import/no-duplicates` is the answer if you want to use inline type specifier.

This is the intended solution and the way that the rules have purposely been designed.

Consistent type imports does exactly as it needs to - it warns against value-only imports. The fixer is a convenience, and supports fixing to top level imports (which satisfies the TS option) or to inline imports.

Your codebase's codestyle is to not have duplicate imports and to merge them if desired using inline specifiers. That style is separately enforceable using `import/no-duplicates`.

We intentionally did not include fixer logic around deduplication for imports because that rules are intended to be single-concern. Mixing too much logic into one rule is not good practice. There are issues where we discussed this and the entire point behind the new rule and rule options added to the import plugins were due to those discussions.

So to summarise - if you want to use this mixed style, use the top level fixer with `import/no-duplicates`
